index.ts

```diff
@@ -1,18 +1,28 @@
 type Options = {
 	filter?: (mutations: MutationRecord[]) => boolean;
+	signal?: AbortSignal;
 } & MutationObserverInit;
 
 export default async function oneMutation(
 	element: Element,
-	options: Options = {},
+	{filter, signal, ...options}: Options = {},
 ): Promise<MutationRecord[]> {
+	if (signal?.aborted) {
+		return [];
+	}
+
 	return new Promise(resolve => {
-		const {filter} = options;
-		new MutationObserver((changes, observer) => {
+		const observer = new MutationObserver(changes => {
 			if (!filter || filter(changes)) {
 				observer.disconnect();
 				resolve(changes);
 			}
-		}).observe(element, options);
+		});
+		observer.observe(element, options);
+
+		signal?.addEventListener('abort', () => {
+			observer.disconnect();
+			resolve([]);
```

Comment:
This promise was somehow not resolving with AVA. `observer.disconnect()` would kill the process silently. 🤷‍♂️ 

Vitest has no such issue